kernel/init.c

```diff
@@ -422,14 +425,15 @@ FUNC_NORETURN void _Cstart(void)
 	 */
 	char dummy_thread_memory[sizeof(struct k_thread)];
 	struct k_thread *dummy_thread = (struct k_thread *)&dummy_thread_memory;
+
+	memset(dummy_thread_memory, 0, sizeof(dummy_thread_memory));
```

Comment:
What do we need to zero out here that we didn't before?
Some initialization of dummy_thread is done at the beginning of prepare_multithreading(), can we consolidate?